File: DheerajKumar/problem07.cpp
//7. WAP to take input 3 digit numbers from the user and print the square of the middle digit.


#include<iostream>
using namespace std;

int middleSquare(int number){
    if(number >= 1000 || number < 100){
        cout<<"out of range !";
        return -1;
    }
    int middle = (number%100)/10;
    return middle * middle;
}

int main(){
    int number;
    cout<< "\n\nEnter the number ->>";
    cin>> number;

    cout<< "\n\n Square of the middle number is -> "<< middleSquare(number);
}

File: LovelyKumari/BasicProgramming/Problem07.java
import java.util.Scanner;
public class Problem07 {
    // WAP to take input 3 digit numbers from the user and print the square of the middle digit.

    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter a 3-digit number: ");
        int num=sc.nextInt();

        //Valid or not 
        if(num>=100 && num<=999){
            int middleDigit=(num/10) % 10;
            int square=middleDigit*middleDigit ;
            System.out.println("The square of the middle digit ("+ middleDigit+ ") = " + square );

        } else {
            System.out.println("Enter a valid 3-digit number.");
        }
            
    }
}

